### PR TITLE
Sarah search fns Merge to Main

### PR DESCRIPTION
**Pull request by spgallant**
Search functionality pulling from TMDB API. Links to search/movie.id and persists search term on refresh and when clicking back from a specific movie detail (result) page.

image

image

image

### CODE REVIEW

**Review by maikkatko — APPROVED** [review]
Nice work! Looks really good

**Review by nancyonyejiaka — CHANGES_REQUESTED** [review]
before you merge:
- clicking on a search result should link to `/film/details/movie.id`

File: src/components/main/Search/index.tsx [hunk 1]
```diff
@@ -100,7 +100,7 @@ export default function Search() {
       <div className="movie-card">
         {searchResults.length > 0 ? (
           searchResults.map((movie: any) => (
-            <Link key={movie._id} to={`/search/${movie.id}`} className="card-link">
+            <Link key={movie._id} to={`/film/details/${movie.id}`} className="card-link">
```
Comment (spgallant) [inline src/components/main/Search/index.tsx:103]: Llink fixed!

**Review by maikkatko — APPROVED** [review]
good to go

**Review by nancyonyejiaka — APPROVED** [review]
nice work!!

**Pull request merged**